test/functionalTest/cases/3190_notify_only_attributes_that_change/notify_only_attributes_that_change.test

```diff
@@ -0,0 +1,497 @@
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print localhost $LISTENER_PORT
+
+--SHELL--
+
+#
+# 01. POST /v2/entities, create Room1 entity with A, B, C, and D attributes.
+# 02. POST /v2/subscriptions, to create subscription for Room1 with onlyChangedAttrs set to true.
+# 03. Dump and reset accumulator, see notification.
+# 04. Sleep 5 secs, before next update.
+# 05. PATCH /v2/entities/Room1/attrs, update the value of A and B attribute.
+# 06. Dump and reset accumulator, see notification.
+# 07. Sleep 5 secs, before next update.
+# 08. PATCH /v2/entities/Room1/attrs, update the value of B and C attribute.
+# 09. Dump and reset accumulator, see notification.
+# 10. Sleep 5 secs, before next update.
+# 11. PATCH /v2/entities/Room1/attrs, update the value of C and D attribute.
+# 12. Dump and reset accumulator, see notification.
+# 13. Sleep 5 secs, before next update.
+# 14. PATCH /v2/entities/Room1/attrs, update the value of A, B, C and D attribute.
+# 15. Dump and reset accumulator, see notification.
+#
+
+echo '01. POST /v2/entities, create Room1 entity with A, B, C, and D attributes.'
+echo '=========================================================================='
+payload='{
+  "id": "Room1",
+  "type": "Room",
+  "A": {
+    "value": 10,
+    "type": "Integer"
+  },
+  "B": {
+    "value": 20,
+    "type": "Integer"
+  },
+  "C": {
+    "value": 30,
+    "type": "Integer"
+  },
+  "D": {
+    "value": 40,
+    "type": "Integer"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+
+echo '02. POST /v2/subscriptions, to create subscription for Room1 with onlyChangedAttrs set to true.'
+echo '==============================================================================================='
+payload='{
+  "description": "A subscription to get info about Room1",
+  "subject": {
+    "entities": [
+      {
+        "id": "Room1",
+        "type": "Room"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "A","B","C"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [
+      "B","C","D"
+    ],
+    "onlyChangedAttrs":true
+  },
+  "expires": "2040-01-01T14:00:00.00Z",
+  "throttling": 5
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo '03. Dump and reset accumulator, see notification.'
+echo '================================================='
+accumulatorDump
+accumulatorReset
+echo
+echo
+
+
+echo '04. Sleep 5 secs, before next update.'
+echo '====================================='
+sleep 5
```

Comment:
Why this inter-step sleep? It is really needed (looking to other similar tests, it doesn't seem so)? In the case of being needed, could it be reduced?

Reply:
Ok, I think I have found the cause. You are using `"throttling": 5`. Please, remove that for subscription (it's not related with the functionality), so you can remove all the sleep steeps from the .test file.

In addition, remove also `"expires"` in the subscription. Ok, 2040 is far away and it doesn't have any practical effect :), but it is not related with the test purpose so let's keep it simple.

Reply:
fixed in f9887f1